utils.js

```diff
@@ -0,0 +1,57 @@
+const Chance = require('chance')
+
+const randomName = () => new Chance().first()
+
+const isFunction = (func) => (func && typeof func === "function")
+
+const arraysIntersection = (...arrays) => {
+    const verifiedArrays = arrays.filter(value => Array.isArray(value))
+    if (arrays.length === 0) return arrays;
+    return verifiedArrays.reduce((acc, currentArray) => {
+        currentArray.forEach(currentValue => {
+            if (acc.indexOf(currentValue) === -1) {
+                if (verifiedArrays.filter((obj) => obj.indexOf(currentValue) === -1).length === 0) {
+                    acc.push(currentValue);
+                }
+            }
+        })
+        return acc;
+    }, [])
+}
```

Comment:
Did you consider 

```js
const fruitsA = ["🍎", "🍌", "🍒"];
const fruitsB = ["🍒", "🍆", "🍉", "🍌"];

const intersection = fruitsA.filter(fruit => fruitsB.includes(fruit));
console.log(intersection); // ["🍌", "🍒"]
```

https://dev.to/kais_blog/14-awesome-javascript-array-tips-you-should-know-about-1d0m#7-how-to-find-the-intersection-of-two-arrays

Reply:
The problem is `lodash` `intersection` is not as simple as that. but I'll try that to check if works